### PR TITLE
Provide option to control log level of oci-umount

### PR DESCRIPTION
**Pull request by rhvgoyal**
Provide an option to control log_level of oci-umount.

### CODE REVIEW

**Comment by rhvgoyal** [comment]
@rhatdan PTAL

**Comment by rhvgoyal** [comment]
@eparis CC

File: oci-umount.spec [hunk 1]
```diff
@@ -54,6 +54,7 @@ make %{?_smp_mflags}
 /usr/share/containers/oci/hooks.d/oci-umount.json
 /usr/share/oci-umount/oci-umount.d
 %ghost /etc/oci-umount/oci-umount.d
+%config /usr/share/oci-umount/oci-umount-options.conf
```
Comment (rhatdan) [inline oci-umount.spec]: This is not config

  %ghost /etc/oci-umount/oci-umount-options.conf

File: src/oci-umount.c [hunk 2]
```diff
@@ -430,6 +457,81 @@ static int is_mount_duplicate(const char*path, struct host_mount_info *mounts_on
 	return 0;
 }
 
+static int parse_options_section_line(const char *id, const char *conf_file, char *line) {
+	int str_len, i;
+	char *level_str;
+
+	str_len = strlen("log_level=");
+
+	if (strncmp(line, "log_level=", str_len)) {
```
Comment (rhatdan) [inline src/oci-umount.c]: You are not handling

   log_level=
log_level = foobar

File: src/oci-umount.c [hunk 3]
```diff
@@ -430,6 +457,81 @@ static int is_mount_duplicate(const char*path, struct host_mount_info *mounts_on
...
+		pr_perror("%s: Unknown option %s in file %s", id, line, conf_file);
+		return EXIT_FAILURE;
+	}
+
+	level_str = line + str_len;
```
Comment (rhatdan) [inline src/oci-umount.c]: This will not handle
log_level= LOG_DEBUG

File: oci-umount.spec (hunk 1 above)
Comment (rhvgoyal) [reply oci-umount.spec]: ok, so config is only used for files which can be edited by user and we don't want to overwrite these over upgrade. This one we don't want users to edit and even if they do, they will lose change over upgrade, so this is not a config. Agreed.

But this does not seem to be `ghost` either? ghost seems to be files which are not shipped with the package. But we are shipping this file with the package?

Comment (rhatdan) [reply oci-umount.spec]: I ghosted the /etc version not the /usr version.

File: src/oci-umount.c (hunk 2 above)
Comment (rhvgoyal) [reply src/oci-umount.c]: should we really support whitespaces around `=`. I am not a big fan of that.

W.r.t `log_level=`, we are getting following error message.

`oci-umount[3906]: umounthook <error>: 10924c826513: Unknown log level  in file /etc/oci-umount/oci-umount-options.conf: Success`

Do you want to display anything else?

File: oci-umount.spec (hunk 1 above)
Comment (rhvgoyal) [reply oci-umount.spec]: aha.. missed that.

Ok, so by making it `ghost`, we will remove this file when package is installed. And that probably is fine.

**Comment by rhvgoyal** [comment]
@rhatdan I have pushed new patches. Fixed spec file. Have not done anything about log_level related comments yet. Have a look. Let me know if you want me to support whitespaces around `=`. Or want me to give some other message for `log_level=` case.

File: src/oci-umount.c (hunk 2 above)
Comment (rhatdan) [reply src/oci-umount.c]: If I am a human being and a computer complains about a space, I get pissed off.  Computers should handle name value pairs with spaces.

Comment (rhvgoyal) [reply src/oci-umount.c]: So where are do you want to support spacing. I find this to be a weak argument. 

- How about section name. `[     options      ]` is it a valid string?
- How about `log_level                 =                   foo`. is it valid
- How about `log_level \t\t = \t\t foo`

I mean there are so many possibilities. And I don't see a strong argument to support those. How hard it is to type, `name=value`. This is so common.

Comment (rhvgoyal) [reply src/oci-umount.c]: github ate away multiple spaces I had put. So we need to support multiple white spaces around `=` as well. And same is true for section  name? 

Does key name has to start from beginning of line? or we need to support white space there to.

Comment (rhatdan) [reply src/oci-umount.c]: Yes I just want spaces trimmed from beginning and end of value.  Nothing too fancy

"log_level = debug  "
Should work.

Comment (rhatdan) [reply src/oci-umount.c]: I would trim spaces in name and value.

**Comment by rhvgoyal** [comment]
@rhatdan Ok, I added to logic to eat all whitespaces from `name=value` line. PTAL.

**Comment by rhvgoyal** [comment]
@rhatdan PTAL. I took care of your comments.

File: src/oci-umount.c [hunk 4]
```diff
@@ -432,6 +457,96 @@ static int is_mount_duplicate(const char*path, struct host_mount_info *mounts_on
...
+	int i, str_len;
+
+	str_len = strlen(str);
+
+	for (i = 0; i < str_len; i++) {
+		if (str[i] == ' ') {
+			memmove(&str[i], &str[i+1], str_len - i + 1);
+		}
+	}
+}
```
[the remark this answers is not in the document]
Comment (cgwalters) [reply src/oci-umount.c]: Is there a reason oci-umount doesn't link to libglib where we have e.g. `g_strchomp()`?

**Comment by rhatdan** [comment]
LGTM

File: src/oci-umount.c (hunk 4 above)
[the remark this answers is not in the document]
Comment (rhatdan) [reply src/oci-umount.c]: Just never built with libglib.

**Pull request merged**